pkg/builtin/unary/oct.go

```diff
@@ -0,0 +1,406 @@
+				results := &types.Bytes{
+					Data:    []byte{},
+					Offsets: make([]uint32, len(col)),
+					Lengths: make([]uint32, len(col)),
+				}
+				results = oct.OctUint64(col, results)
+
+				resVec, err := process.Get(proc, int64(len(results.Data)), types.Type{Oid: types.T_varchar, Size: 24})
+				if err != nil {
+					return nil, err
+				}
+				nulls.Set(resVec.Nsp, origVec.Nsp)
+				vector.SetCol(resVec, results)
+				return resVec, nil
+			},
+		},
+		{ // T_int8 to T_char
+			Typ:        types.T_int8,
+			ReturnType: types.T_char,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				col := origVec.Col.([]int8)
+				results := &types.Bytes{
+					Data:    []byte{},
+					Offsets: make([]uint32, len(col)),
+					Lengths: make([]uint32, len(col)),
+				}
+				results = oct.OctInt8(col, results)
+
+				resVec, err := process.Get(proc, int64(len(results.Data)), types.Type{Oid: types.T_char, Size: 24})
+				if err != nil {
+					return nil, err
+				}
+				nulls.Set(resVec.Nsp, origVec.Nsp)
+				vector.SetCol(resVec, results)
+				return resVec, nil
+			},
+		},
+		{ // T_int8 to T_varchar
+			Typ:        types.T_int8,
+			ReturnType: types.T_varchar,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				col := origVec.Col.([]int8)
+				results := &types.Bytes{
+					Data:    []byte{},
+					Offsets: make([]uint32, len(col)),
+					Lengths: make([]uint32, len(col)),
+				}
+				results = oct.OctInt8(col, results)
+
+				resVec, err := process.Get(proc, int64(len(results.Data)), types.Type{Oid: types.T_varchar, Size: 24})
+				if err != nil {
+					return nil, err
+				}
+				nulls.Set(resVec.Nsp, origVec.Nsp)
+				vector.SetCol(resVec, results)
+				return resVec, nil
+			},
+		},
+		{ // T_int16 to T_char
+			Typ:        types.T_int16,
+			ReturnType: types.T_char,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				col := origVec.Col.([]int16)
+				results := &types.Bytes{
+					Data:    []byte{},
+					Offsets: make([]uint32, len(col)),
+					Lengths: make([]uint32, len(col)),
+				}
+				results = oct.OctInt16(col, results)
+
+				resVec, err := process.Get(proc, int64(len(results.Data)), types.Type{Oid: types.T_char, Size: 24})
+				if err != nil {
+					return nil, err
+				}
+				nulls.Set(resVec.Nsp, origVec.Nsp)
+				vector.SetCol(resVec, results)
+				return resVec, nil
+			},
+		},
+		{ // T_int16 to T_varchar
+			Typ:        types.T_int16,
+			ReturnType: types.T_varchar,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				col := origVec.Col.([]int16)
+				results := &types.Bytes{
+					Data:    []byte{},
+					Offsets: make([]uint32, len(col)),
+					Lengths: make([]uint32, len(col)),
+				}
+				results = oct.OctInt16(col, results)
+
+				resVec, err := process.Get(proc, int64(len(results.Data)), types.Type{Oid: types.T_varchar, Size: 24})
```

Comment:
is the results.Data's length fixed? I don't know, for int16 ---> (16 % 3 + 1) * numberOfRows?

if we find a way to allocate the right size of memory for result.Data field, then we can get rid of the append(result.Data).... code later in our function execution, I think get rid of this append thing may help boost this function's performance

Reply:
I think the length may not be fixed. For example, a negative integer is expected to convert into a 64bit number like 
```MySQL
mysql> SELECT OCT(-128);
        -> '1777777777777777777600'
```
in MySQL. However positive integer is flexible.

I also approve that append(result.Data).... code is not the most efficient approach. A potential way is supposed to use 22*numberOfRows where 22 is the longest width for an integer. It wastes some space but we can return Data[:offset[last]+length[last]]. Traversing the original integers to judge length I think may not high-efficiency.